### PR TITLE
[portsorch] Configure hostif tagging for subports

### PR DESCRIPTION
**Pull request by vsenchyshyn**
Signed-off-by: Vitaliy Senchyshyn <vsenchyshyn@barefootnetworks.com>

**What I did**
Configure SAI_HOSTIF_VLAN_TAG_KEEP for the parent port hostif when a first subport is added and restore it to SAI_HOSTIF_VLAN_TAG_STRIP when the last subport is removed.

**Why I did it**
As the parent port hostif tagging is not modified during subports creation the packets destined to the subport interfaces are forwarded as untagged by the packet driver and therefore never reaching the subport netdevs.

**How I verified it**

Created two subports using the config below and pinged their IPs with vlan 10 and 20 tagged packets respectively. The ping was successful.

```
{
    "VLAN_SUB_INTERFACE": {
        "Ethernet1.10": {
            "admin_status" : "up"
        },
        "Ethernet1.10|11.0.10.1/24": {},

        "Ethernet1.20": {
            "admin_status" : "up"
        },
        "Ethernet1.20|11.0.20.1/24": {}
    }
}
``` 

**Details if related**

### CODE REVIEW

**Comment by vsenchyshyn** [comment]
retest vs please

**Comment by vsenchyshyn** [comment]
@prsunny @stcheng @lguohan Could you please review

**Comment by vsenchyshyn** [comment]
retest vs please

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -738,6 +749,21 @@ bool PortsOrch::removeSubPort(const string &alias)
     m_portList[parentPort.m_alias] = parentPort;
 
     m_portList.erase(it);
+
+    // Restore hostif vlan tag for the parent port when the last subport is removed
+    if (parentPort.m_child_ports.empty())
+    {
+        if (!parentPort.m_bridge_port_id)
+        {
+            if (!setHostIntfsStripTag(port, SAI_HOSTIF_VLAN_TAG_STRIP))
```
Comment (prsunny) [inline orchagent/portsorch.cpp:758]: Is it port or parentPort?

**Comment by prsunny** [comment]
request @wendani to review

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (vsenchyshyn) [reply orchagent/portsorch.cpp]: Good catch! Fixed.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -738,6 +749,21 @@ bool PortsOrch::removeSubPort(const string &alias)
...
     m_portList.erase(it);
+
+    // Restore hostif vlan tag for the parent port when the last subport is removed
+    if (parentPort.m_child_ports.empty())
+    {
+        if (!parentPort.m_bridge_port_id)
+        {
+            if (!setHostIntfsStripTag(parentPort, SAI_HOSTIF_VLAN_TAG_STRIP))
+            {
+                SWSS_LOG_WARN("Failed to set %s for hostif of port %s",
```
Comment (wendani) [inline orchagent/portsorch.cpp:760]: What is the thought for `WARN` instead of `ERROR` as above?

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -738,6 +749,21 @@ bool PortsOrch::removeSubPort(const string &alias)
     m_portList[parentPort.m_alias] = parentPort;
 
     m_portList.erase(it);
+
+    // Restore hostif vlan tag for the parent port when the last subport is removed
+    if (parentPort.m_child_ports.empty())
+    {
+        if (!parentPort.m_bridge_port_id)
```
Comment (wendani) [inline orchagent/portsorch.cpp:756]: `== SAI_NULL_OBJECT_ID`

**Comment by wendani** [comment]
Can you also validate on the lag parent port case?

**Comment by wendani** [comment]
Looks relevant to https://github.com/Azure/sonic-buildimage/issues/3943

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (vsenchyshyn) [reply orchagent/portsorch.cpp]: Copy & paste from the log right above in the function. Didn't notice that there is no return false.

File: orchagent/portsorch.cpp (hunk 3 above)
Comment (vsenchyshyn) [reply orchagent/portsorch.cpp]: Done

**Comment by vsenchyshyn** [comment]
> Can you also validate on the lag parent port case?

I'm not sure I can do this. I've applied the following config on a switch running t0 topo config:

{
    "VLAN_SUB_INTERFACE": {
        "PortChannel0001.10": {
            "admin_status" : "up"
        },
        "PortChannel0001.10|11.0.10.1/24": {},

        "PortChannel0001.20": {
            "admin_status" : "up"
        },
        "PortChannel0001.20|11.0.20.1/24": {}
    }
}

But this didn't have any effect - no subinterface PortChannel0001.10 created in linux. The config present in config db only.

**Comment by wendani** [comment]
> > Can you also validate on the lag parent port case?
> 
> I'm not sure I can do this. I've applied the following config on a switch running t0 topo config:
> 
> ```
> {
>     "VLAN_SUB_INTERFACE": {
>         "PortChannel0001.10": {
>             "admin_status" : "up"
>         },
>         "PortChannel0001.10|11.0.10.1/24": {},
> 
>         "PortChannel0001.20": {
>             "admin_status" : "up"
>         },
>         "PortChannel0001.20|11.0.20.1/24": {}
>     }
> }
> ```
> 
> But this didn't have any effect - no subinterface PortChannel0001.10 created in linux. The config present in config db only.
Keep port channel name within 15 characters. Rename to `PortChannel1.10`

**Comment by OleksandrKozodoi** [comment]
> Can you also validate on the lag parent port case?

@wendani I've validated this case. 
**Steps for validation:**
1. Deploy t0 topology.
2. Deploy minigrapgh on DUT
3. Remove member port from Vlan1000 on the DUT
```
config vlan member del 1000 Ethernet8
```
4. Create portchannel on the DUT
```
config portchannel add PortChannel5 --fallback=true
```
5. Add member port to portchannel on the DUT
```
config portchannel member add PortChannel5 Ethernet8
```
6. Setup bond interface on the PTF
```
ip link add bond2 type bond
ip link set bond2 type bond miimon 100 mode balance-xor
ip link set eth2 down
ip link set eth2 master bond2
ip link set dev bond2 up
ifconfig bond2 hw ether 22:22:22:22:22:21
ifconfig bond2 mtu 9216 up
```
7. Setup configuration of sub-ports on the DUT
**_sub_port_config.j2_:**
```
{
	"VLAN_SUB_INTERFACE": {

		"PortChannel5.10": {
			"admin_status" : "up"
		},
		"PortChannel5.10|172.16.0.5/30": {}
	}
}
```
```
sonic-cfggen -j sub_port_config.j2 --write-to-db
```
8. Setup configuration of sub-ports on the PTF
```
ip link add link bond2 name bond2.10 type vlan id 10
ip address add 172.16.0.6/30 dev bond2.10
ip link set bond2.10 up
```
9. Ping DUT from PTF
``` 
ping -I bond2.10 172.16.0.5
```
**Validation results**
```
/# ping -I bond2.10 172.16.0.5
PING 172.16.0.5 (172.16.0.5) from 172.16.0.6 bond2.10: 56(84) bytes of data.
64 bytes from 172.16.0.5: icmp_seq=1 ttl=64 time=0.405 ms
64 bytes from 172.16.0.5: icmp_seq=2 ttl=64 time=0.445 ms
64 bytes from 172.16.0.5: icmp_seq=3 ttl=64 time=0.771 ms
64 bytes from 172.16.0.5: icmp_seq=4 ttl=64 time=0.746 ms
```
```
~$ tcpdump -nei PortChannel5.10
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on PortChannel5.10, link-type EN10MB (Ethernet), capture size 262144 bytes
14:55:19.045230 22:22:22:22:22:21 > 74:83:ef:c9:b4:38, ethertype IPv4 (0x0800), length 98: 172.16.0.6 > 172.16.0.5: ICMP echo request, id 911, seq 1862, length 64
14:55:19.045305 74:83:ef:c9:b4:38 > 22:22:22:22:22:21, ethertype IPv4 (0x0800), length 98: 172.16.0.5 > 172.16.0.6: ICMP echo reply, id 911, seq 1862, length 64
14:55:20.069271 22:22:22:22:22:21 > 74:83:ef:c9:b4:38, ethertype IPv4 (0x0800), length 98: 172.16.0.6 > 172.16.0.5: ICMP echo request, id 911, seq 1863, length 64
14:55:20.069359 74:83:ef:c9:b4:38 > 22:22:22:22:22:21, ethertype IPv4 (0x0800), length 98: 172.16.0.5 > 172.16.0.6: ICMP echo reply, id 911, seq 1863, length 64
14:55:21.093466 22:22:22:22:22:21 > 74:83:ef:c9:b4:38, ethertype IPv4 (0x0800), length 98: 172.16.0.6 > 172.16.0.5: ICMP echo request, id 911, seq 1864, length 64
14:55:21.093536 74:83:ef:c9:b4:38 > 22:22:22:22:22:21, ethertype IPv4 (0x0800), length 98: 172.16.0.5 > 172.16.0.6: ICMP echo reply, id 911, seq 1864, length 64
```

**Review by wendani — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**